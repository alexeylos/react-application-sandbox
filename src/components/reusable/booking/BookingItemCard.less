.passenger-name {
  text-decoration: underline;
  padding-bottom: 4px;
}

.booking-dttm {
  margin-bottom: 15px;
  font-size: 13px;
}

.departure,
.arrival {
  display: flex;
  flex-direction: column;
  .station-name {
    font-weight: 600;
    font-size: 13px;
  }
  .time-date {
    font-size: 12px;
  }
}

.currency {
  font-size: 15px;
  font-weight: 600;
  margin-top: 15px;
}
