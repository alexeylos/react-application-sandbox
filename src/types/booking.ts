export interface Booking {
  id: string;
  booking_dttm: string;
  carrier_name: string;
  passenger: string;
  departure_station: string;
  departure_dttm: string;
  arrival_station: string;
  arrival_dttm: string;
  total_price: string;
  currency: string;
  status: string;
}
