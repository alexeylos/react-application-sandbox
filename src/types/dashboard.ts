export interface DashboardData {
  user_name: string;
  tickets_sold: number;
}
