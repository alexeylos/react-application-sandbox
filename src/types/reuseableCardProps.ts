export interface ReusableCardProps {
  title: string;
  value: number;
  delta: number;
  currency: string;
}

export interface FeatureCardProps {
  icon: React.ReactNode;
  title: string;
  description: string;
  buttonLink?: string;
}
