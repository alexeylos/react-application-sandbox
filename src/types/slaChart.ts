export interface IndicatorValue {
  value: number;
  date: string;
  category?: string;
}

export interface Indicator {
  id: string;
  name: string;
  type: string;
  title: string;
  unit: string;
  description: string;
  internal_objective: number;
  external_objective: number;
  values: IndicatorValue[];
}

export interface Agreement {
  id: string;
  name: string;
  indicators: Indicator[];
}
