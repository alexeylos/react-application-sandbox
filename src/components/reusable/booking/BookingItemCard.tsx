import { Booking } from '@/types/booking';
import { ArrowDownOutlined, ClockCircleOutlined } from '@ant-design/icons';
import { Card, Col, Row, Tag } from 'antd';
import dayjs from 'dayjs';
import React, { useCallback } from 'react';
import { useNavigate } from 'react-router-dom';
import './BookingItemCard.less';

const BookingItemCard: React.FC<Booking> = ({
  id,
  booking_dttm,
  carrier_name,
  passenger,
  departure_station,
  departure_dttm,
  arrival_station,
  arrival_dttm,
  total_price,
  currency,
  status,
}) => {
  const navigate = useNavigate();
  const onClickNavigate = useCallback(() => {
    navigate(`/bookings/${id}`);
  }, [id]);

  return (
    <>
      <Card onClick={onClickNavigate}>
        <Row gutter={[0, 16]}>
          <Col span={18}>
            <p className="passenger-name">
              <strong>{passenger}</strong>
            </p>
            <p className="booking-dttm">
              <ClockCircleOutlined />
              {`${dayjs(booking_dttm).format('YYYY-MM-DD HH:mm')}, ${carrier_name}`}
            </p>
            <div>
              <div className="departure">
                <span className="station-name">{departure_station}</span>
                <span className="time-date">
                  {dayjs(departure_dttm).format('YYYY-MM-DD HH:mm')}
                </span>
              </div>

              <div>
                <ArrowDownOutlined />
              </div>

              <div className="arrival">
                <span className="station-name">{arrival_station}</span>
                <span className="time-date">{dayjs(arrival_dttm).format('YYYY-MM-DD HH:mm')}</span>
              </div>
            </div>

            <div className="currency">
              <span>
                {total_price} {currency}
              </span>
            </div>
          </Col>

          <Col span={6}>
            <Row justify="end">
              <Tag color={status === 'active' ? 'green' : 'red'}>
                {(status ?? 'unknown').toUpperCase()}
              </Tag>
            </Row>
          </Col>
        </Row>
      </Card>
    </>
  );
};

export default BookingItemCard;
