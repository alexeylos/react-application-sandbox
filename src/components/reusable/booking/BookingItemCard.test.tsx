import { Booking } from '@/types/booking';
import { render, screen } from '@testing-library/react';
import dayjs from 'dayjs';
import BookingItemCard from './BookingItemCard';

const booking: Booking = {
  id: 'B1',
  booking_dttm: '2024-10-14T22:32:00Z',
  carrier_name: 'Carrier 1',
  passenger: 'Passenger 1',
  departure_station: 'Station 1',
  departure_dttm: '2024-10-14T22:32:00Z',
  arrival_station: 'Station 2',
  arrival_dttm: '2024-10-14T23:32:00Z',
  total_price: '100.00',
  currency: 'USD',
  status: 'active',
};

describe('BookingItemCard', () => {
  it('renders BookingItemCard correctly', () => {
    render(<BookingItemCard {...booking} />);
    expect(screen.getByText(booking.passenger)).toBeInTheDocument();
    expect(screen.getByText(new RegExp(booking.carrier_name, 'i'))).toBeInTheDocument();
    expect(screen.getByText(`${booking.total_price} USD`)).toBeInTheDocument();
    expect(screen.getByText(booking.status.toUpperCase())).toBeInTheDocument();
  });

  it('displays formatted departure and arrival times correctly', () => {
    render(<BookingItemCard {...booking} />);
    expect(
      screen.getByText(dayjs(booking.departure_dttm).format('YYYY-MM-DD HH:mm')),
    ).toBeInTheDocument();
    expect(
      screen.getByText(dayjs(booking.arrival_dttm).format('YYYY-MM-DD HH:mm')),
    ).toBeInTheDocument();
  });

  it('displays the correct status color based on status', () => {
    render(<BookingItemCard {...booking} />);
    const statusTag = screen.getByText(booking.status.toUpperCase());
    expect(statusTag).toHaveClass('ant-tag-green');

    const cancelledBooking = { ...booking, status: 'cancelled' };
    render(<BookingItemCard {...cancelledBooking} />);
    expect(screen.getByText('CANCELLED')).toHaveClass('ant-tag-red');
  });

  it('renders total price with currency', () => {
    render(<BookingItemCard {...booking} />);
    expect(screen.getByText(`${booking.total_price} ${booking.currency}`)).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<BookingItemCard {...booking} />);
    expect(asFragment()).toMatchSnapshot();
  });
});

it('renders BookingItemCard correctly', () => {
  render(<BookingItemCard {...booking} />);
  expect(screen.getByText(booking.passenger)).toBeInTheDocument();
  expect(screen.getByText(new RegExp(booking.carrier_name, 'i'))).toBeInTheDocument();
  expect(screen.getByText(`${booking.total_price} USD`)).toBeInTheDocument();
  expect(screen.getByText(booking.status.toUpperCase())).toBeInTheDocument();
});
