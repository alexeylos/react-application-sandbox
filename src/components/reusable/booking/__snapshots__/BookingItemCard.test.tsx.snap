// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookingItemCard matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="ant-card ant-card-bordered css-dev-only-do-not-override-m4timi"
  >
    <div
      class="ant-card-body"
    >
      <div
        class="ant-row css-dev-only-do-not-override-m4timi"
        style="row-gap: 16px;"
      >
        <div
          class="ant-col ant-col-18 css-dev-only-do-not-override-m4timi"
        >
          <p
            class="passenger-name"
          >
            <strong>
              Passenger 1
            </strong>
             
          </p>
          <p
            class="booking-dttm"
          >
            <span
              aria-label="clock-circle"
              class="anticon anticon-clock-circle"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="clock-circle"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                />
                <path
                  d="M686.7 638.6L544.1 535.5V288c0-4.4-3.6-8-8-8H488c-4.4 0-8 3.6-8 8v275.4c0 2.6 1.2 5 3.3 6.5l165.4 120.6c3.6 2.6 8.6 1.8 11.2-1.7l28.6-39c2.6-3.7 1.8-8.7-1.8-11.2z"
                />
              </svg>
            </span>
             2024-10-15 04:32, Carrier 1
          </p>
          <div
            class=""
          >
            <div
              class="departure"
            >
              <span
                class="station-name"
              >
                Station 1
              </span>
              <span
                class="time-date"
              >
                2024-10-15 04:32
              </span>
            </div>
            <div
              class=""
            >
              <span
                aria-label="arrow-down"
                class="anticon anticon-arrow-down"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="arrow-down"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M862 465.3h-81c-4.6 0-9 2-12.1 5.5L550 723.1V160c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8v563.1L255.1 470.8c-3-3.5-7.4-5.5-12.1-5.5h-81c-6.8 0-10.5 8.1-6 13.2L487.9 861a31.96 31.96 0 0048.3 0L868 478.5c4.5-5.2.8-13.2-6-13.2z"
                  />
                </svg>
              </span>
            </div>
            <div
              class="arrival"
            >
              <span
                class="station-name"
              >
                Station 2
              </span>
              <span
                class="time-date"
              >
                2024-10-15 05:32
              </span>
            </div>
          </div>
          <div
            class="currency"
          >
            <span>
              100.00 USD
            </span>
          </div>
        </div>
        <div
          class="ant-col ant-col-6 css-dev-only-do-not-override-m4timi"
        >
          <div
            class="ant-row ant-row-end css-dev-only-do-not-override-m4timi"
          >
            <span
              class="ant-tag ant-tag-green css-dev-only-do-not-override-m4timi"
            >
              ACTIVE
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
